modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -923,6 +943,12 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
             node = std::make_shared<ngraph::op::v1::GreaterEqual>(inp0, inp1);
         else if (op == OPERATION::LESS_EQUAL)
             node = std::make_shared<ngraph::op::v1::LessEqual>(inp0, inp1);
+        // Ideally we should do this but int32 internal blobs are converted to float32 data type in inference.
+        // FIXME: enable this when internal blobs support int32 data type
+        // else if (op == OPERATION::IMOD)
+        //     node = std::make_shared<ngraph::op::v1::FloorMod>(inp0, inp1);
```

Comment:
Maybe `ngraph::op::Convert` can help here?

Reply:
It should work. Let me have a try on this.

Reply:
Done.